src/libServer/Server.cpp

```diff
@@ -504,7 +520,12 @@ Json::Value Server::GetSmartContracts(const string& address) {
       throw JsonRpcException(RPC_INVALID_PARAMETER,
                              "Address size not appropriate");
     }
-    bytes tmpaddr = DataConversion::HexStrToUint8Vec(address);
+    bytes tmpaddr;
+    if (!DataConversion::HexStrToUint8Vec(address, tmpaddr)) {
+      throw JsonRpcException(RPC_INVALID_ADDRESS_OR_KEY,
+                             "address is not a hex string");
```

Comment:
you may want to use the same message `invalid address` that's used at other places